cheroot/test/test_ssl.py

```diff
@@ -696,6 +694,92 @@ def test_https_over_http_error(http_server, ip_addr):
     assert expected_substring in ssl_err.value.args[-1]
 
 
+@pytest.mark.parametrize(  # noqa: C901  # FIXME
+    'adapter_type',
+    (
+        'builtin',
+        'pyopenssl',
+    ),
+)
+def test_plain_http_check_no_data(
+    adapter_type,
+    tls_certificate_chain_pem_path,
+    tls_certificate_private_key_pem_path,
+    mocker,
+):
+    """Test ``_check_for_plain_http()`` handles empty peek correctly."""
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+
+    tls_adapter = tls_adapter_cls(
+        tls_certificate_chain_pem_path,
+        tls_certificate_private_key_pem_path,
+    )
+
+    mock_socket = mocker.MagicMock()
+    mock_socket.recv.return_value = b''  # Empty peek
+    mock_socket.gettimeout.return_value = None
+
+    result = tls_adapter._check_for_plain_http(mock_socket)
+
+    assert result is False
+    mock_socket.recv.assert_called_once_with(16, socket.MSG_PEEK)
+
+
+@pytest.mark.parametrize(
+    'adapter_type',
+    ('builtin', 'pyopenssl'),
+)
+def test_plain_http_check_socket_error(
```

Comment:
Looks like the exception side effect in this test function and the next one can be parametrized to avoid duplication (it's a 99% copy-paste except for one line, right?)